### PR TITLE
Updates to match changes in Nengo

### PR DESCRIPTION
**Pull request by bmorcos**
**Motivation and context:**

Nengo has made some breaking changes, this brings NengoFPGA back into compatibility.

**How has this been tested?**

- Tested with basic exampe and notebooks with full stack.
- Have not tested GUI yet (pending GUI patch)

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- Bug fix (non-breaking change which fixes an issue)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- N/A I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [x] I have tested this with all supported devices.
- N/A I have added tests to cover my changes.
- N/A I have run the test suite locally and all tests passed.

**Still to do:**

- [ ] Test GUI examples

### CODE REVIEW

**Comment by bmorcos** [comment]
TravisCI can't pip install Nengo>=3.0.0... maybe we just let this sit here until the 3.0 release?

**Comment by bmorcos** [comment]
Similarly, `nengo-gui` needed to be patched for compat. and `nengo_extras`...

**Comment by bmorcos** [comment]
Having some trouble getting the (compatible) GUI to log info...

File: CHANGES.rst [hunk 1]
```diff
@@ -26,6 +26,10 @@ Release History
 - Setup Nengo Bones and remote CI.
   (`#41 <https://github.com/nengo/nengo-fpga/pull/41>`__)
 
+**Changed**
```
Comment (bmorcos) [inline CHANGES.rst]: Should we put this under the **Fixed ** header instead maybe?

Comment (xchoo) [reply CHANGES.rst]: Changed is okay. But the changelog should specify the changes were made for Nengo 3.0.0.

**Comment by xchoo** [comment]
Why is this PR complaining about conflicting files?

**Comment by bmorcos** [comment]
Yea, I think this is just saying it can't automatically merge from the GUI since there are conflicts (this is a commit behind `master`. So yea, should all be fixed in the manual merge I imagine.

**Comment by bmorcos** [comment]
I patched the URLs for now, but I made a note to start using proper intersphinx references.

**Comment by bmorcos** [comment]
I patched those URLs also, did I miss something?

**Pull request merged**